### PR TITLE
Ref bootstrapping (disclaimer)

### PR DESCRIPTION
**Pull request by cmgustavo**

### CODE REVIEW

**Comment by cmgustavo** [comment]
Updated

File: src/js/controllers/disclaimer.js [hunk 1]
```diff
@@ -35,13 +23,16 @@ angular.module('copayApp.controllers').controller('disclaimerController',
...
+      self.lang = uxLanguage.currentLanguage;
+      storageService.getProfile(function(err, profile) {
+        if (!profile) create();
+        else $scope.creatingProfile = false;
 
-      //compatible
-      storageService.getCopayDisclaimerFlag(function(err, val) {
-        if (val) go.walletHome();
+        //compatible
+        profileService.checkDisclaimer(function(val) {
```
Comment (matiu) [inline src/js/controllers/disclaimer.js]: name suggestion: isDisclaimerAccepted

File: src/js/controllers/index.js [hunk 2]
```diff
@@ -156,6 +148,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     self.usingCustomBWS = config.bwsFor && config.bwsFor[self.walletId] && (config.bwsFor[self.walletId] != defaults.bws.url);
   };
 
+  self.agreeDisclaimer = function() {
```
Comment (matiu) [inline src/js/controllers/index.js]: name suggestion: acceptDisclaimer

File: src/js/controllers/index.js [hunk 3]
```diff
@@ -156,6 +148,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
+    var profile = profileService.profile;
+    if (profile) profile.agreeDisclaimer = true;
```
Comment (matiu) [inline src/js/controllers/index.js]: name suggestion: profile.disclaimerAccepted = true

File: src/js/controllers/index.js [hunk 4]
```diff
@@ -156,6 +148,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
+    self.disclaimerAccepted = true;
+    profileService.storeDisclaimer(function(err) {
```
Comment (matiu) [inline src/js/controllers/index.js]: name suggestion: profileservice.setDisclaimerAccepted

File: src/js/controllers/index.js [hunk 5]
```diff
@@ -156,6 +148,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
+      go.walletHome();
+    });
+  };
+
+  self.checkDisclaimer = function() {
+    if (self.disclaimerAccepted == true) { 
+      go.walletHome();
+      return;
+    }
+    profileService.checkDisclaimer(function(v) {
```
Comment (matiu) [inline src/js/controllers/index.js]: isDisclaimerAccepted

File: src/js/controllers/index.js [hunk 6]
```diff
@@ -156,6 +148,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
+    var profile = profileService.profile;
+    if (profile) profile.agreeDisclaimer = true;
+    self.disclaimerAccepted = true;
+    profileService.storeDisclaimer(function(err) {
+      if (err) $log.error(err); 
+      go.walletHome();
+    });
+  };
+
+  self.checkDisclaimer = function() {
```
Comment (matiu) [inline src/js/controllers/index.js]: isDisplaimerAccepted

**Comment by matiu** [comment]
Tested OK. LGTM.

**Comment by cmgustavo** [comment]
Updated. Thanks for the comments!

**Pull request merged**